languages/python/oso/polar/query.py

```diff
@@ -88,47 +86,33 @@ def handle_make_external(self, data):
 
     def handle_external_call(self, data):
         call_id = data["call_id"]
-        if call_id not in self.calls:
-            value = data["instance"]["value"]
-            instance = self.host.to_python(data["instance"])
-
-            attribute = data["attribute"]
-
-            # Lookup the attribute on the instance.
-            try:
-                attr = getattr(instance, attribute)
-            except AttributeError as e:
-                self.ffi_query.application_error(str(e))
-                self.ffi_query.call_result(call_id, None)
-                return
-            if (
-                callable(attr) and not data["args"] is None
-            ):  # If it's a function, call it with the args.
-                args = [self.host.to_python(arg) for arg in data["args"]]
-                kwargs = data["kwargs"] or {}
-                kwargs = {k: self.host.to_python(v) for k, v in kwargs.items()}
-                result = attr(*args, **kwargs)
-            elif not data["args"] is None:
-                raise InvalidCallError(
-                    f"tried to call '{attribute}' but it is not callable"
-                )
-            else:  # If it's just an attribute, it's the result.
-                result = attr
-
-            # We now have either a generator or a result.
-            # Call must be a generator so we turn anything else into one.
-            if type(result) in NATIVE_TYPES or not isinstance(result, Iterable):
-                call = (i for i in [result])
-            else:
-                call = iter(result)
-            self.calls[call_id] = call
+        instance = self.host.to_python(data["instance"])
 
-        # Return the next result of the call.
+        attribute = data["attribute"]
+
+        # Lookup the attribute on the instance.
         try:
-            value = next(self.calls[call_id])
-            self.ffi_query.call_result(call_id, self.host.to_polar(value))
-        except StopIteration:
+            attr = getattr(instance, attribute)
+        except AttributeError as e:
+            self.ffi_query.application_error(str(e))
             self.ffi_query.call_result(call_id, None)
+            return
+        if (
+            callable(attr) and not data["args"] is None
```

Comment:
I know this isn't related to your changes, but... 
So `data["args"] is None` is how we check whether or not this was parsed as a field or method? If it's a method with no args, is it an empty list? Should we add an explicit flag here, rather than relying on the `None` check?

Reply:
It's especially weird for that to be the check for methods, since we now have `kwargs`, which will be `None` if you don't have `kwargs`, even if the call is a method.

Reply:
> I know this isn't related to your changes, but... 
> So `data["args"] is None` is how we check whether or not this was parsed as a field or method? If it's a method with no args, is it an empty list? Should we add an explicit flag here, rather than relying on the `None` check?

All of this is correct.

- Yes that's how we check
- Yes empty list for call instead of none
- a flag might be better. You would still want/need to check the none/list case but might make it less error prone?

Reply:
Yeah, would also improve readability.

Reply:
If its okay, I would prefer to deal with this in a follow up API cleanup.

Reply:
Cool, made a notion task